destination/destination_integration_test.go

```diff
@@ -0,0 +1,349 @@
+	}
+
+	t.Run("destination open success", func(t *testing.T) {
+		is := is.New(t)
+
+		d := NewDestination()
+
+		ctx := context.Background()
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:       "localhost:2222",
+			config.ConfigHostKey:       hostKey,
+			config.ConfigUsername:      "user",
+			config.ConfigPassword:      "pass",
+			config.ConfigDirectoryPath: "/upload",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.NoErr(err)
+
+		d.Teardown(ctx)
+	})
+
+	t.Run("destination open error sshConfig", func(t *testing.T) {
+		is := is.New(t)
+
+		d := NewDestination()
+
+		ctx := context.Background()
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:       "localhost:2222",
+			config.ConfigHostKey:       "hostKey",
+			config.ConfigUsername:      "user",
+			config.ConfigPassword:      "pass",
+			config.ConfigDirectoryPath: "/upload",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.True(err != nil)
+		is.Equal(err.Error(), "failed to create SSH config: failed to parse server public key: ssh: no key found")
+
+		d.Teardown(ctx)
+	})
+
+	t.Run("destination open error read private key", func(t *testing.T) {
+		is := is.New(t)
+
+		d := NewDestination()
+
+		ctx := context.Background()
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:        "localhost:2222",
+			config.ConfigHostKey:        hostKey,
+			config.ConfigUsername:       "user",
+			config.ConfigPrivateKeyPath: "privatekey",
+			config.ConfigDirectoryPath:  "/upload",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.True(err != nil)
+		is.Equal(err.Error(), "failed to create SSH config: failed to read private key file: open privatekey: no such file or directory")
+
+		d.Teardown(ctx)
+	})
+
+	t.Run("destination open error ssh dial", func(t *testing.T) {
+		is := is.New(t)
+
+		d := NewDestination()
+
+		ctx := context.Background()
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:       "localhost:22",
+			config.ConfigHostKey:       hostKey,
+			config.ConfigUsername:      "root",
+			config.ConfigPassword:      "root",
+			config.ConfigDirectoryPath: "/home/root",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.True(err != nil)
+		is.Equal(err.Error(), "failed to dial SSH: ssh: handshake failed: host key type mismatch: got ecdsa-sha2-nistp256, want ssh-rsa")
+	})
+
+	t.Run("destination open error remote path not exist", func(t *testing.T) {
+		is := is.New(t)
+
+		d := NewDestination()
+
+		ctx := context.Background()
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:       "localhost:2222",
+			config.ConfigHostKey:       hostKey,
+			config.ConfigUsername:      "user",
+			config.ConfigPassword:      "pass",
+			config.ConfigDirectoryPath: "/home/root",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.True(err != nil)
+		is.Equal(err.Error(), "remote path does not exist: file does not exist")
+	})
+}
+
+func TestDestination_Write(t *testing.T) {
```

Comment:
It would be really good if we had a test for chunking too.

Reply:
Added.